src/combinatorics/simple_graph/ends/defs.lean

```diff
@@ -0,0 +1,233 @@
+/-
+Copyright (c) 2022 Anand Rao, Rémi Bottinelli. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anand Rao, Rémi Bottinelli
+-/
+import combinatorics.simple_graph.basic
+import combinatorics.simple_graph.connectivity
+import data.set_like.basic
+import category_theory.category.basic
+import category_theory.filtered
+import topology.category.Top.limits
+
+/-!
+# Ends
+
+This file contains a definition of the ends of a simple graph, as sections of the inverse system
+assigning, to each finite set of vertices, the connected components of its complement.
+-/
+
+universes u
+variables {V : Type u} (G : simple_graph V) (K L L' M : set V)
+
+open classical
+
+noncomputable theory
+local attribute [instance] prop_decidable
```

Comment:
I just noticed this line -- any reason it's not the more standard `open_locale classical`?

Reply:
We actually don't need _any_ of this!